action.go

```diff
@@ -182,6 +192,46 @@ func (c *Connection) Print(path string, clNumber int) ([]byte, error) {
 	}
 }
 
+// GetAllChangelists gets changelists starting from n-th changelist and return a ChangeList slice.
+func (c *Connection) GetAllChangelists(depot string, changeFrom int) ([]ChangeList, error) {
+	changelists := []ChangeList{}
+	if data, err := c.execP4("changes", "-e", strconv.Itoa(changeFrom), depot); err == nil {
+		// Might want to limit this later on for mem safety
+		submatch := getAllChangelistsRegexp.FindAllSubmatch(data, -1)
+		for _, changelist := range submatch {
+			intChange, err := strconv.Atoi(string(changelist[1]))
+			if err != nil {
+				return nil, err
```

Comment:
It could be useful to wrap this error to give a bit more context when this fails. I've run into this myself where I cannot debug the issue right away as I don't know what the input was that actually caused the error and the internal error doesn't display it sometimes.
Therefore it can be useful to include the changelist-string itself in the error to speed up debugging.
Same goes for the dateChange below.

Reply:
But this might just be my preference :)

Reply:
Added a bit more context to these errors in 6aec1c4